api/src/main/java/org/datacleaner/configuration/RemoteServerData.java

```diff
@@ -33,4 +33,8 @@
     public String getUsername();
 
     public String getPassword();
+
+    public void setUsername(String username);
```

Comment:
Each remote server provider has its data. In my case, when password is changed. I can do changes only in this object, and call refresh on descriptor provider. Everything is working.

But in your case. I will must find actual remote data. Remove it. Create new data, and after that set this data to right provider. Call refresh on providers.

If you wish, i will do this change, but my code is simple and elegant.